Description: Disable wxWidgets uncaught exception handling
 wxWidgets' exception handling gets in the way of debugging by
 handling uncaught exceptions, thous hiding their point of origin.
 This causes the build to fail:
 application.cpp:249:2: error: #error why is wxWidgets uncaught exception handling enabled!?
  249 | #error why is wxWidgets uncaught exception handling enabled!?
      |  ^~~~~
 compilation terminated due to -Wfatal-errors.
Author: Jhonny Oliveira <jhonny.oliveira@gmail.com>
Bug: https://freefilesync.org/forum/viewtopic.php?t=10794
Forwarded: not-needed
Last-Update: 2023-10-29
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/FreeFileSync/Source/RealTimeSync/application.cpp
+++ b/FreeFileSync/Source/RealTimeSync/application.cpp
@@ -213,9 +213,6 @@
 
 int Application::OnRun()
 {
-#if wxUSE_EXCEPTIONS
-#error why is wxWidgets uncaught exception handling enabled!?
-#endif
 
     //exception => Windows: let it crash and create mini dump!!! Linux/macOS: std::exception::what() logged to console
         [[maybe_unused]] const int rc = wxApp::OnRun();
--- a/FreeFileSync/Source/application.cpp
+++ b/FreeFileSync/Source/application.cpp
@@ -239,9 +239,6 @@
 
 int Application::OnRun()
 {
-#if wxUSE_EXCEPTIONS
-#error why is wxWidgets uncaught exception handling enabled!?
-#endif
 
     //exception => Windows: let it crash and create mini dump!!! Linux/macOS: std::exception::what() logged to console
         [[maybe_unused]] const int rc = wxApp::OnRun();
